modules/dxp/apps/saml/saml-persistence-service/src/main/java/com/liferay/saml/persistence/internal/upgrade/v3_0_0/SamlSpSessionUpgradeProcess.java

```diff
@@ -38,35 +39,54 @@ protected void doUpgrade() throws Exception {
 					new AlterTableAddColumn("samlPeerBindingId", "LONG null"));
 			}
 
-			int samlSpSessionIdOffset = 0;
+			int samlSpSessionIdOffset = _getSamlSpSessionIdOffset();
+
+			int latestSamlPeerBindingId = _getLatestSamlPeerBindingId();
 
 			try (PreparedStatement preparedStatement =
 					connection.prepareStatement(
-						"select min(samlSpSessionId) - 1 from SamlSpSession");
+						StringBundler.concat(
+							"select min(samlIdpSpSessionId) as ",
+							"samlIdpSpSessionId, companyId, min(createDate) ",
+							"as createDate, userId, userName, nameIdFormat, ",
+							"nameIdNameQualifier, nameIdValue, samlSpEntityId ",
+							"from SamlIdpSpSession group by companyId, ",
```

Comment:
This UpgradeProcess is for `SamlSpSession`. There is a separate one from `SamlIdpSpSession` which we can apply the same general approach to.

Reply:
OMG, I was reviewing the whole structure and I think I messed up with the copy & paste, good catch!
Fixed 😅